gradle-palantir-java-format/src/main/java/com/palantir/javaformat/gradle/spotless/PalantirJavaFormatStep.java

```diff
@@ -51,25 +51,36 @@ static final class State implements Serializable {
 
         // Kept for state serialization purposes.
         @SuppressWarnings({"unused", "FieldCanBeLocal"})
-        private final FileSignature jarsSignature;
+        private FileSignature jarsSignature;
+
+        private final transient Supplier<Iterable<File>> jarsSupplier;
 
         // Transient as this is not serializable.
         private final transient Supplier<FormatterService> memoizedFormatter;
 
         /**
          * Build a cacheable state for spotless from the given jars, that uses the given {@link FormatterService}.
          *
-         * @param jars The jars that contain the palantir-java-format implementation. This is only used for caching and
+         * @param jarsSupplier Supplies the jars that contain the palantir-java-format implementation. This is only used for caching and
          * up-to-dateness purposes.
          */
-        State(Iterable<File> jars, Supplier<FormatterService> memoizedFormatter) throws IOException {
-            this.jarsSignature = FileSignature.signAsSet(jars);
+        State(Supplier<Iterable<File>> jarsSupplier, Supplier<FormatterService> memoizedFormatter) {
+            this.jarsSupplier = jarsSupplier;
             this.memoizedFormatter = memoizedFormatter;
         }
 
         @SuppressWarnings("NullableProblems")
         FormatterFunc createFormat() {
-            return memoizedFormatter.get()::formatSourceReflowStringsAndFixImports;
+            return input -> {
+                try {
+                    // Only resolve the jars and compute the signature at execution time!
+                    Iterable<File> jars = jarsSupplier.get();
+                    this.jarsSignature = FileSignature.signAsSet(jars);
```

Comment:
I think it's worth figuring out how often this is called. It's the only thing that will be repeatedly called, and jars can be big, so you don't want to spend a ton of time doing jar hashing repeatedly and unnecessarily

Reply:
Running spotlessApply on the whole project with multiple modified files only runs `createFormat` once, so I think there's no perf concern here. 

Also, hashing is needed to construct the State so I don't think we can get rid of it here.

Reply:
I wasn't suggesting getting rid of hashing, just not doing it repeatedly i.e. how many times is `createFormat` called. 

I think there's a separate issue where we create this state once per *project*, so the more projects you have, you create a `FormatterFunc` and therefore repeatedly hash the contents. This exists before this change and this change doesn't make it worse. In the context of build perf, it's probably worth making this go away. I would mark this as a FLUP for this PR.